File: sdk/typescript/examples/basic-usage.ts
/**
 * Basic usage example for Loopai TypeScript SDK
 */

import { LoopaiClient } from '../src';

async function main() {
  // Initialize client
  const client = new LoopaiClient({
    baseUrl: 'http://localhost:8080',
  });

  try {
    // Check health
    const health = await client.getHealth();
    console.log(`API Status: ${health.status} (v${health.version})`);

    // Create a task
    const task = await client.createTask({
      name: 'spam-classifier',
      description: 'Classify emails as spam or not spam',
      inputSchema: {
        type: 'object',
        properties: {
          text: { type: 'string' },
        },
        required: ['text'],
      },
      outputSchema: {
        type: 'string',
        enum: ['spam', 'not_spam'],
      },
      accuracyTarget: 0.95,
      latencyTargetMs: 50,
    });
    console.log(`\nCreated task: ${task.name} (${task.id})`);

    // Execute single input
    const result = await client.execute({
      taskId: task.id,
      input: { text: 'Buy now for free money!' },
    });
    console.log(
      `\nClassification: ${result.output} (latency: ${result.latencyMs}ms)`
    );

    // Batch execute
    const emails = [
      { text: 'Buy now for free money!' },
      { text: 'Meeting tomorrow at 2pm' },
      { text: 'URGENT: Click here NOW!!!' },
      { text: 'Can we schedule a call next week?' },
    ];

    const batchResult = await client.batchExecute(task.id, emails, 10);

    console.log('\nBatch Results:');
    console.log(`  Total: ${batchResult.totalItems}`);
    console.log(`  Success: ${batchResult.successCount}`);
    console.log(`  Failed: ${batchResult.failureCount}`);
    console.log(`  Avg latency: ${batchResult.avgLatencyMs.toFixed(2)}ms`);
    console.log(`  Total duration: ${batchResult.totalDurationMs.toFixed(2)}ms`);

    console.log('\nIndividual Results:');
    for (const item of batchResult.results) {
      if (item.success) {
        console.log(
          `  ${item.id}: ${item.output} (latency: ${item.latencyMs.toFixed(2)}ms, ` +
            `sampled: ${item.sampledForValidation})`
        );
      } else {
        console.log(`  ${item.id}: FAILED - ${item.errorMessage}`);
      }
    }
  } catch (error) {
    console.error('Error:', error);
  }
}

main();

File: sdk/python/examples/basic_usage.py
"""Basic usage example for Loopai Python SDK."""

import asyncio
from loopai import LoopaiClient


async def main() -> None:
    """Demonstrate basic SDK usage."""
    # Initialize client
    async with LoopaiClient("http://localhost:8080") as client:
        # Check health
        health = await client.get_health()
        print(f"API Status: {health.status} (v{health.version})")

        # Create a task
        task = await client.create_task(
            name="spam-classifier",
            description="Classify emails as spam or not spam",
            input_schema={
                "type": "object",
                "properties": {"text": {"type": "string"}},
                "required": ["text"],
            },
            output_schema={"type": "string", "enum": ["spam", "not_spam"]},
            accuracy_target=0.95,
            latency_target_ms=50,
        )
        print(f"Created task: {task.name} ({task.id})")

        # Execute single input
        result = await client.execute(
            task_id=task.id, input_data={"text": "Buy now for free money!"}
        )
        print(f"Classification: {result.output} (latency: {result.latency_ms}ms)")

        # Batch execute
        emails = [
            {"text": "Buy now for free money!"},
            {"text": "Meeting tomorrow at 2pm"},
            {"text": "URGENT: Click here NOW!!!"},
            {"text": "Can we schedule a call next week?"},
        ]

        batch_result = await client.batch_execute(task.id, emails, max_concurrency=10)

        print(f"\nBatch Results:")
        print(f"  Total: {batch_result.total_items}")
        print(f"  Success: {batch_result.success_count}")
        print(f"  Failed: {batch_result.failure_count}")
        print(f"  Avg latency: {batch_result.avg_latency_ms:.2f}ms")
        print(f"  Total duration: {batch_result.total_duration_ms:.2f}ms")

        print(f"\nIndividual Results:")
        for item in batch_result.results:
            if item.success:
                print(
                    f"  {item.id}: {item.output} (latency: {item.latency_ms:.2f}ms, "
                    f"sampled: {item.sampled_for_validation})"
                )
            else:
                print(f"  {item.id}: FAILED - {item.error_message}")


if __name__ == "__main__":
    asyncio.run(main())
